Source/Global/global.cpp

```diff
@@ -15,42 +15,45 @@ using namespace xbox::httpclient;
 
 NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
 
-enum class singleton_access_mode
+enum singleton_access_mode
 {
     create,
     get,
     cleanup
 };
 
-HRESULT singleton_access(
+HRESULT http_singleton::singleton_access(
     _In_ singleton_access_mode mode,
     _In_opt_ HCInitArgs* createArgs,
     _Out_ std::shared_ptr<http_singleton>& singleton
 ) noexcept
 {
     static std::mutex s_mutex;
     static std::shared_ptr<http_singleton> s_singleton{ nullptr };
+    static uint8_t s_useCount{ 0 };
 
     std::lock_guard<std::mutex> lock{ s_mutex };
     switch (mode)
     {
     case singleton_access_mode::create:
     {
-        if (s_singleton)
+        // Create the singleton only for the first client calling create
+        if (!s_useCount++)
```

Comment:
could you write it as
```c++
if (s_useCount == 0)
{
    ...
}
++s_useCount;

Reply:
In fact right now you potentially increment the use count before creating the singleton, which could lead to a situation where we have use count > 0 but no object